### DISCUSSION TITLE
Potential generator compatibility issue.

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#10773] [General]
I just want to log an issue I encountered when I was trying to develop Avalonia apps in a new machine.
This machine has only .NET 6.0.114 installed, and I have no rights to upgrade it. When I upgrade to Avalonia preview 6, it shows below errors: 

Warning	CS8032	An instance of analyzer Avalonia.Generators.NameGenerator.AvaloniaNameSourceGenerator cannot be created from C:\Users\***\.nuget\packages\avalonia\11.0.0-preview6\analyzers\dotnet\cs\Avalonia.Generators.dll: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified..	Assistant		1	Active	Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.

Seems to be roslyn backward comaptibility issue, but I don't know how to fix it.

**Comment by maxkatz6** [comment]
Pretty sure 4.3.0 version should work instead. But I haven't tried it yet.

**Comment by rabbitism** [comment]
@maxkatz6 for now I disabled the built-in generator, and added XamlNameReferenceGenerator back. but for backward compatibility concern, we should check if it's possible to downgrade to 3.8 (which is used by Avalonia.Diagnostic)

**Reply by maxkatz6** [reply]
Old Avalonia.NameGenerator.csproj used 3.9. I am going to downgrade to the same version.

There is no problems with Diagnostic package, as user application doesn't directly reference analyzers package, so it won't conflict.

**Reply by DominicMaas** [reply]
How did you disable the built in generator? I tried adding below and referenced the old package, but still get the same error.
```xml
<AvaloniaNameGeneratorIsEnabled>false</AvaloniaNameGeneratorIsEnabled>
```

**Reply by kekekeks** [reply]
`ExcludeAssets="analyzers"`
